edk2toollib/uefi/edk2/parsers/base_parser_test.py

```diff
@@ -385,6 +388,27 @@ def test_process_conditional_reset(self):
         self.assertTrue(parser.InActiveCode())
         self.assertEqual(len(parser.ConditionalStack), 0)
 
+    def test_process_in_conditional(self):
+        parser = BaseParser("")
+        parser.SetInputVars({"TOOL_CHAIN_TAG": "GCC5_TEST"})
+        self.assertTrue(parser.ProcessConditional(
+            '!if ("GCC49" in $(TOOL_CHAIN_TAG)) OR ("GCC5" in $(TOOL_CHAIN_TAG))'))
+        self.assertTrue(parser.InActiveCode())
+        parser.ResetParserState()
```

Comment:
Are there any new tests that exploit the added functionality? Tests that would have failed previously but now pass?

Reply:
Did not mean to approve.

Reply:
you can see that this is a new test that would have failed on the old parser but it now works. However, we really don't have that many tests that really flex the muscles of the new conditional parser.

Reply:
There's an issue that tracks this: #74

Reply:
I added a few more test but more are needed. Our test coverage for base parser is pretty great (95% last time I checked) but we need more of the failure cases.

Reply:
Okay further tweaks and added a few more tests